src/pages/Trade/components/ChartCard/Chart.tsx

```diff
@@ -29,22 +41,132 @@ type Props = {
 	period: PeriodLabel;
 	synthPair: SynthPair;
 };
-type ChartProps = StateProps & Props;
 
-const Chart: FC<ChartProps> = ({ synthPair: { quote }, data, isLoading, period, synthsMap }) => {
+type ChartProps = StateProps & DispatchProps & Props;
+
+const Chart: FC<ChartProps> = ({
+	walletInfo,
+	fetchMyTradesRequest,
+	trades,
+	synthPair: { quote },
+	data,
+	isLoading,
+	period,
+	synthsMap,
+}) => {
 	const { colors } = useContext(ThemeContext);
 	const { t } = useTranslation();
 	const synthSign = synthsMap[quote.name] && synthsMap[quote.name].sign;
 
+	useEffect(() => {
+		fetchMyTradesRequest();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [walletInfo.currentWallet]);
+
+	useInterval(() => {
+		fetchMyTradesRequest();
+	}, DEFAULT_REQUEST_REFRESH_INTERVAL);
+
+	const ratesLength = (data?.rates ?? []).length;
+	const tradesLength = (trades ?? []).length;
+
+	const dataWithTrades = useMemo(() => {
+		if (trades.length > 0 && data.rates.length > 0 && walletInfo.currentWallet != null) {
+			const newRates = data.rates.map((rate, index) => {
+				const trade = find(
+					trades,
+					(trade: HistoricalTrade) =>
+						rate.block < trade.block &&
+						index < data.rates.length - 1 &&
+						data.rates[index + 1].block > trade.block
+				);
+
+				return trade
+					? {
+							...rate,
+							tradeTotal: trade.fromAmountInUSD,
+					  }
+					: rate;
+			});
+
+			return {
+				...data,
+				rates: newRates,
+			};
+		}
+		return data;
+	}, [ratesLength, tradesLength, walletInfo.currentWallet]);
+
+	const CustomizedDot = ({
+		cx,
+		cy,
+		payload,
+	}: {
+		cx: number;
+		cy: number;
+		payload: {
+			tradeTotal?: number;
+		};
+	}) => {
+		if (payload.tradeTotal) {
+			return (
+				<svg x={cx - 5} y={cy - 5} width="6" height="15" viewBox="0 0 6 15" fill="none">
+					<rect width="6" height="15" rx="1" fill={colors.fontSecondary} />
+				</svg>
+			);
+		}
+
+		return null;
+	};
+
+	const CustomTooltip = ({
+		active,
+		label,
+		payload,
+	}: {
+		active: boolean;
+		payload: [
+			{
+				value: number;
+				payload: {
+					tradeTotal?: number;
+				};
+			}
+		];
+		label: Date;
+	}) => {
+		if (active) {
+			const { value, payload: innerPayload } = payload[0];
+			return (
+				<TooltipContentStyle>
+					<LabelStyle>{format(label, 'do MMM yy | HH:mm')}</LabelStyle>
+					<ItemStyle>{`${t('trade.chart-tooltip.price')}: ${synthSign}${formatCurrencyWithPrecision(
+						value
+					)}`}</ItemStyle>
+					{innerPayload.tradeTotal ? (
+						<ItemStyle>{`${t('trade.chart-tooltip.trade-total')}: $${formatCurrencyWithPrecision(
```

Comment:
Use `${synthSign}` - btw you can use `formatCurrencyWithSign(sign, value)` and it will do the same... sorry we got a bit of mess with the formatters.